core/extract_data/core/nixpkgs-graph.nix

```diff
@@ -0,0 +1,116 @@
+# usage:
+# NIXPKGS_ALLOW_BROKEN=1 NIXPKGS_ALLOW_INSECURE=1 TARGET_FLAKE_REF="github:nixos/nixpkgs/master" TARGET_SYSTEM="x86_64-linux" nix eval --json --file "./nixpkgs-graph.nix"
+
+let
+  nixpkgs = builtins.getFlake "nixpkgs";
+  targetFlakeRef = builtins.getEnv "TARGET_FLAKE_REF";
+  # We specify the system of the target here to avoid ambiguity
+  # The outputs of a flake will produce different results on every system
+  targetSystem = builtins.getEnv "TARGET_SYSTEM";
+  targetFlakePkgs =
+    let
+      flake = builtins.getFlake targetFlakeRef;
+    in
+      flake.outputs.packages.${targetSystem} or flake.outputs.defaultPackage.${targetSystem} or flake.outputs.legacyPackages.${targetSystem} or { };
+in
+
+with nixpkgs.lib;
+
+let
+  recurse =
+    name: value:
+    let
+      valueEvalResult = builtins.tryEval value;
+    in
+    if valueEvalResult.success then
+      let okValue = valueEvalResult.value;
+      in
+      if isDerivation okValue then
+        {
+          # inherit path;
+          # can't name it `outPath` because serialization would only output it instead of dict
+          # see Nix `toString` docs
+          # we don't query pname and version directly because some derivations only have a `name` attribute.
+          # use `builtins.parseDrvName` to get the name on all cases
+
+          name = (builtins.tryEval (if okValue ? name then okValue.name else "")).value;
+          output_paths =
+            let
+              outputs =
+                let
+                  pEvalResult = builtins.tryEval (toString (if okValue ? outputs
+                                                            then okValue.outputs else ""));
+                in
+                if pEvalResult.success then splitString " " pEvalResult.value
+                else [ ];
+            in
+            map
+              (p:
+                { name = p;
+                  path = (builtins.tryEval (if okValue ? ${p}
+                                            then toString okValue.${p} else "")).value; }
+              )
+              (outputs);
+          nixpkgs_metadata =
+            {
+              pname = (builtins.tryEval (if okValue ? name && okValue.name != false
```

Comment:
`nixpkgs_metadata.pname` should really be the `pname` of the package attrset, and null if none

Reply:
if we want to include data that we parse, we should put that into another data block to make expectations clear